src/app/auth/signup/index.tsx

```diff
@@ -238,68 +223,28 @@ function SignUpScreen() {
         </View>
 
         {password !== '' && (
-          <View style={styles.passwordComplexity}>
-            <Icon type={hasUppercase ? 'green_check' : 'grey_dot'} />
-            <Text
-              style={[
-                globalStyles.errorMessage,
-                styles.passwordErrorText,
-                hasUppercase
-                  ? { color: colors.textGreen }
-                  : { color: colors.textGrey },
-              ]}
-            >
-              At least 1 uppercase letter
-            </Text>
-          </View>
+          <PasswordComplexityText
+            condition={hasUppercase}
+            message="At least 1 uppercase letter"
+          ></PasswordComplexityText>
```

Comment:
Just a small nitpick here since everything else is looking good: Since there's nothing inside this component you can do this instead:
```
<PasswordComplexityText
    condition={hasUppercase}
    message="At least 1 uppercase letter"
/>
```
where you don't need that whole closing tag and you just add the closing slash. Just a small nit to make the code a little cleaner!